File: app/src/main/java/com/github/gfx/helium/widget/LoadingIndicatorViewHolder.java
package com.github.gfx.helium.widget;

import com.github.gfx.helium.util.LoadingAnimation;

import android.content.Context;
import android.databinding.ViewDataBinding;
import android.support.annotation.LayoutRes;
import android.support.annotation.NonNull;
import android.view.ViewGroup;

public class LoadingIndicatorViewHolder<T extends ViewDataBinding> extends BindingHolder<T> {

    final LoadingAnimation loadingAnimation = new LoadingAnimation();

    public LoadingIndicatorViewHolder(@NonNull Context context, @NonNull ViewGroup parent,
            @LayoutRes int resource) {
        super(context, parent, resource);

        loadingAnimation.start(binding.getRoot());
    }
}

File: app/src/main/java/com/github/gfx/helium/widget/LoadingIndicatorViewHolder.kt
package com.github.gfx.helium.widget

import com.github.gfx.helium.util.LoadingAnimation

import android.content.Context
import android.databinding.ViewDataBinding
import android.support.annotation.LayoutRes
import android.view.ViewGroup

class LoadingIndicatorViewHolder<T : ViewDataBinding>(context: Context, parent: ViewGroup,
                                                      @LayoutRes resource: Int) : BindingHolder<T>(context, parent, resource) {

    internal val loadingAnimation = LoadingAnimation()

    init {
        loadingAnimation.start(binding.root)
    }
}
